tests/unit_tests/test_semantic_tools.py

```
@@ -0,0 +1,220 @@
+        assert "original_rows" in compressed_data
+        assert "original_columns" in compressed_data
+        assert "is_compressed" in compressed_data
+        assert "compressed_data" in compressed_data
+        assert "removed_columns" in compressed_data
+        assert "compression_type" in compressed_data
+
+        # Verify metadata is preserved
+        assert result_dict["columns"] == ["date", "revenue", "orders"]
+        assert result_dict["metadata"] == {"execution_time": 0.5}
+
+    def test_query_metrics_small_data_not_compressed(self, semantic_tools, mock_adapter):
```

Ruff (0.15.0)

[warning] Unused method argument: `mock_adapter`

(ARG002)